contracts/colonyNetwork/ColonyNetwork.sol

```diff
@@ -167,12 +220,57 @@ contract ColonyNetwork is BasicMetaTransaction, ColonyNetworkStorage, Multicall
       }
     } else {
       // Add a global skill
+      require(isMiningChain(), "colony-network-not-mining-chain");
       s.globalSkill = true;
-      skills[skillCount] = s;
+      skills[_skillId] = s;
     }
+  }
 
-    emit SkillAdded(skillCount, _parentSkillId);
-    return skillCount;
+  function addSkillFromBridge(uint256 _parentSkillId, uint256 _skillId) public always onlyMiningChain() {
+    // Require is a known bridge
+    Bridge storage bridge = bridgeData[msgSender()];
+    require(bridge.chainId != 0, "colony-network-not-known-bridge");
+
+    // Check skill count - if not next, then store for later.
+    if (networkSkillCounts[bridge.chainId] + 1 == _skillId){
+      if (_parentSkillId > bridge.chainId << 128){
+        addSkillToChainTree(_parentSkillId, _skillId);
+      }
+      networkSkillCounts[bridge.chainId] += 1;
+      emit SkillAdded(_skillId, _parentSkillId);
```

Comment:
Another hangover from trying to incorporate global skills, which wouldn't have gone through the `addSkillToChainTree` path. But now they're not here, again, I see no reason why this can't be done.